src/components/dashboard/Links.tsx

```diff
@@ -28,11 +28,53 @@ const Links = () => {
 
   const filteredResources = useMemo(() => {
     if (searchQuery) {
-      return fuse.search(searchQuery).map((result) => result.item);
+      const fuseResults = fuse.search(searchQuery).map((result) => result.item);
+      const fuseRanking = fuseResults.map((item, index) => ({
+        item,
+        rankFuse: index + 1,
+      }));
+
+      const preferenceRanking = [...fuseResults].sort((a, b) => {
+        const numA = Number(localStorage.getItem(a.url)) || a.frequency;
+        const numB = Number(localStorage.getItem(b.url)) || b.frequency;
```

Comment:
Better to have one localStorage key where the frequencies are stored in an object